compiler/rustc_hir_analysis/src/check/method/suggest.rs

```diff
@@ -2324,6 +2324,61 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 
+    /// issue #102320, for `unwrap_or` with closure as argument, suggest `unwrap_or_else`
+    /// FIXME: currently not working for suggesting `map_or_else`, see #102408
+    pub(crate) fn suggest_else_fn_with_closure(
+        &self,
+        err: &mut Diagnostic,
+        expr: &hir::Expr<'_>,
+        found: Ty<'tcx>,
+        can_satisfy: impl FnOnce(Ty<'tcx>) -> bool,
```

Comment:
This doesn't need to be an argument. Can you inline the `can_coerce` call here by passing in `expected: Ty<'tcx>`?